app/UiTPAS/Listeners/SendMailForUdbOrganizer.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\UiTPAS\Listeners;
+
+use App\Domain\Contacts\ContactType;
+use App\Domain\Integrations\Events\UdbOrganizerCreated;
+use App\Domain\Integrations\Repositories\IntegrationRepository;
+use App\Domain\Integrations\Repositories\UdbOrganizerRepository;
+use App\Domain\Mail\Mailer;
+use App\Domain\UdbUuid;
+use App\Mails\Smtp\MailerTemplate;
+use App\Search\Sapi3\SearchService;
+use App\Search\UdbOrganizerNameResolver;
+use App\UiTPAS\Event\UdbOrganizerApproved;
+use App\UiTPAS\Event\UdbOrganizerRejected;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Contracts\Routing\UrlGenerator as UrlGeneratorContract;
+use Ramsey\Uuid\UuidInterface;
+use Symfony\Component\Mime\Address;
+
+final class SendMailForUdbOrganizer implements ShouldQueue
+{
+    use Queueable;
+
+    public function __construct(
+        private readonly Mailer $mailer,
+        private readonly UdbOrganizerRepository $udbOrganizerRepository,
+        private readonly IntegrationRepository $integrationRepository,
+        private readonly UdbOrganizerNameResolver $udbOrganizerNameResolver,
+        private readonly SearchService $searchService,
+        private readonly UrlGeneratorContract $urlGenerator,
+        private readonly Address $from,
+    ) {
+    }
+
+    public function handleUdbOrganizerCreated(UdbOrganizerCreated $event): void
+    {
+        $udbOrganizer = $this->udbOrganizerRepository->getById($event->id);
+        $this->sendMail(
+            $udbOrganizer->organizerId,
+            $udbOrganizer->integrationId,
+            MailerTemplate::ORGANISATION_UITPAS_REQUESTED->value
+        );
+    }
+
+    public function handleUdbOrganizerApproved(UdbOrganizerApproved $event): void
+    {
+        $this->sendMail($event->udbId, $event->integrationId, MailerTemplate::ORGANISATION_UITPAS_APPROVED->value);
+    }
+
+    public function handleUdbOrganizerRejected(UdbOrganizerRejected $event): void
+    {
+        // Be careful here, at this point the UdbOrganizer is deleted in the db, which is why we sent both the udbId and the integrationId
```

Comment:
How can the mailer delete the organizer?

Reply:
I guess this comment is a bit out of context here?

Reply:
Move the comment to the top of the event. I tripped over it during development, which is why I added the warning.